src/GraphODataTemplateWriter/CodeHelpers/Java/TypeHelperJava.cs

```diff
@@ -1165,15 +1185,35 @@ public static string CreatePackageDef(this CustomT4Host host)
 
             // determine current namespace and generate method imports if applicable
             string @namespace;
-            string methodImports = string.Empty;
+            var methodImports = new HashSet<string>();
+            const string importFormat = "import {0}.{1}.{2};";
+            const string graphServiceEntityName = "GraphService";
+            const string interfaceTemplatePrefix = "I";
             switch (host.CurrentType)
             {
                 case OdcmProperty p:
                     @namespace = p.Type.Namespace.Name.AddPrefix();
+                    if (p.Class.GetTypeString() != graphServiceEntityName)
+                        methodImports.Add(string.Format(importFormat, p.Class.Namespace.Name.AddPrefix(), p.Class.GetPackagePrefix(), p.Class.GetTypeString()));
+                    if (!(p.Projection.Type is OdcmPrimitiveType))
+                        methodImports.Add(string.Format(importFormat, p.Projection.Type.Namespace.Name.AddPrefix(), p.Projection.Type.GetPackagePrefix(), p.Projection.Type.GetTypeString()));
+                    p.Projection?.Type?.AsOdcmClass()?.MethodsAndOverloads()?.Distinct()?.SelectMany(o => ImportClassesOfMethodParameters(o))?.ToList()?.ForEach(x => methodImports.Add(x));
                     break;
                 case OdcmMethod m:
-                    var sb = new StringBuilder(Environment.NewLine);
-                    methodImports = ImportClassesOfMethodParameters(m, "import {0}.{1}.{2};", sb).ToString();
+                    m.WithDistinctOverloads().SelectMany(o => ImportClassesOfMethodParameters(o))?.ToList()?.ForEach(x => methodImports.Add(x));
+                    goto default;
+                case OdcmClass c:
+                    if (c.GetTypeString() != graphServiceEntityName)
+                        methodImports.Add(string.Format(importFormat, c.Namespace.Name.AddPrefix(), c.GetPackagePrefix(), c.GetTypeString()));
```

Comment:
style: do you mind having curly brackets for single-line blocks as well to keep consistency in the code base? There are a few instances of this style but it is mostly styled as blocks.

Reply:
I personally find it makes the syntax heavy to add curlys for single line blocks. Indentation should suffice from a readability perspective. Then again if it's a standard across the team, happy to oblige.

Reply:
I have seen bugs originating from someone forgetting to turn it into a block while adding another statement and somehow making its way into production, so it is also a safeguard measure for that potential error to be missed in code reviews.

+1 on less syntax, python is one of my favorite languages in terms of readability.